app/controllers/bots_controller.rb

```diff
@@ -64,6 +64,83 @@ def destroy
       format.json { head :no_content }
     end
   end
+  
+  
+  # POST /bots/1/collaborators
+  # POST /bots/1/collaborators.json
+  def add_collaborator
+      unless Bot.exists?(id: params[:bot])
```

Comment:
Looks like one too many indent levels here - Rails indents are 2 spaces, which you've got perfect everywhere else in the file, but the body of `add_collaborator` is on a 4-space indent.